paper-api/src/main/java/io/papermc/paper/event/entity/EntityEquipmentChangedEvent.java

```diff
@@ -0,0 +1,66 @@
+package io.papermc.paper.event.entity;
+
+import java.util.Collections;
+import java.util.Map;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.entity.EntityEvent;
+import org.bukkit.inventory.EquipmentSlot;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Unmodifiable;
+import org.jspecify.annotations.NullMarked;
+
+/**
+ * Called whenever a change to an entity's equipment has been detected. This event is called after effects from
+ * attribute modifiers and enchantments have been updated.
+ */
+@NullMarked
+public class EntityEquipmentChangedEvent extends EntityEvent {
```

Comment:
It might be worth documenting here that this event is also called upon entities being added to worlds, since it isn't immediately obvious to users of the event that the game counts that as "changed".

Reply:
Added some examples of situations that can call the event.